### PR TITLE
Add chart color generation

### PR DESCRIPTION
**Pull request by ranquild**
Epic https://github.com/metabase/metabase/issues/22816

Changes:
- Adds a way to automatically fill in missing chart colors
- Changes the palette preview to show 3 bars with different tints/shades combinations
- Adjusts the way how tints and shades are generated to prevent generating colors that are too light

How to test:
- Run Metabase EE
- Go to Admin > Settings > Appearance
- Reset all charts colors to default and click `Generate chart colors`. It should set all main accents.
- Reset all charts colors to default and fill in some of them, but not all. Clicking on `Generate` should set missing values based on the first not empty color.
- Make sure that after generating colors all manually set tints (light variants) and shades (dark variants) are cleared out.

Screenshot 2022-06-03 at 17 04 59

### CODE REVIEW

File: enterprise/frontend/src/metabase-enterprise/whitelabel/components/ChartColorSettings/utils.ts [hunk 1]
```diff
@@ -1,4 +1,59 @@
...
+  const newHue = (color.hue() + newHueChange) % 360;
+  const newSaturation = newHue <= 65 || newHue >= 345 ? 55 : 40;
+
+  return color
+    .hue(newHue)
+    .saturationv(newSaturation)
+    .value(90);
+};
+
+const isCloseColor = (newColor: Color, oldColor: Color) => {
```
[the remark this answers is not in the document]
Comment (ranquild) [reply enterprise/frontend/src/metabase-enterprise/whitelabel/components/ChartColorSettings/utils.ts]: Thanks, changed

**Comment by mazameli** [comment]
I'm seeing a behavior where when I input a hex value, than leave the input, what I entered gets cleared out: https://www.loom.com/share/1bc844ab85e149f4831b0c83f478affa

**Comment by mazameli** [comment]
Here's one value I entered that results in a too-light tint: 

image

image

**Comment by ranquild** [comment]
> I'm seeing a behavior where when I input a hex value, than leave the input, what I entered gets cleared out: https://www.loom.com/share/1bc844ab85e149f4831b0c83f478affa

It's because the string has no `#` at the start

**Comment by ranquild** [comment]
> Here's one value I entered that results in a too-light tint:
> 
> image
> 
> image

Changed a bit how tints are generated, this issue is fixed.

**Review by alxnddr — APPROVED** [review]

**Pull request merged**